core-bundle/src/Resources/contao/classes/StyleSheets.php

```diff
@@ -1387,10 +1387,18 @@ public function exportStyleSheet(DataContainer $dc)
 		$objFile = new \File('system/tmp/' . md5(uniqid(mt_rand(), true)));
 		$objFile->write('');
 
+		$blnClose = false;
+
 		// Add the media query (see #7560)
 		if ($objStyleSheet->mediaQuery != '')
 		{
 			$objFile->append('@media ' . $objStyleSheet->mediaQuery . ' {');
+			$blnClose = true;
+		}
+		elseif (\is_array(($tmp = \StringUtil::deserialize($objStyleSheet->media))) && (\count($tmp) > 1 || $tmp[0] != 'all'))
```

Comment:
How can this ever happen if the previous `if` condition checks for a non-empty field?

Reply:
The previous condition checks for `->mediaQuery`, which is the text field where you can enter a media query. The second condition checks for `->media`, which is the checkbox wizard where you can pick one or more media types.